client/py/synnax/control/controller.py

```diff
@@ -116,11 +117,18 @@ def set(
             channels = retrieve_required(self._retriever, list(ch.keys()))
             now = TimeStamp.now()
             updated = {channels[i].key: values[i] for i in range(len(channels))}
-            updated_idx = {channels[i].index: now for i in range(len(channels))}
+            updated_idx = {
+                channels[i].index: now
+                for i in range(len(channels))
+                if channels[i].virtual != True
```

Comment:
`if not channels[i].virtual` would be cleaner.

Reply:
Fixed